internal/xds/translator/translator.go

```diff
@@ -86,28 +87,34 @@ func (t *Translator) Translate(ir *ir.Xds) (*types.ResourceVersionTable, error)
 	// to fail the entire xDS translation to panic users, but instead, we want
 	// to collect all errors and reflect them in the status of the CRDs.
 	var errs error
+	listenerExtPolicies := make(listenerHookPolicies)
+
 	if err := t.processHTTPListenerXdsTranslation(
-		tCtx, ir.HTTP, ir.AccessLog, ir.Tracing, ir.Metrics); err != nil {
+		tCtx, xdsIR.HTTP, xdsIR.AccessLog, xdsIR.Tracing, xdsIR.Metrics, listenerExtPolicies); err != nil {
+		errs = errors.Join(errs, err)
+	}
+
+	if err := t.processTCPListenerXdsTranslation(tCtx, xdsIR.TCP, xdsIR.AccessLog, xdsIR.Metrics, listenerExtPolicies); err != nil {
 		errs = errors.Join(errs, err)
 	}
 
-	if err := processTCPListenerXdsTranslation(tCtx, ir.TCP, ir.AccessLog, ir.Metrics); err != nil {
+	if err := processUDPListenerXdsTranslation(tCtx, xdsIR.UDP, xdsIR.AccessLog, xdsIR.Metrics, listenerExtPolicies); err != nil {
```

Comment:
looks like we are passing `listenerExtPolicies` to `processUDPListenerXdsTranslation` so that we can append per listener type's `ExtensionRefs` to it. Can we instead move that out of the pure xds translations into an extension specific function like `notifyExtensionServerAboutListeners` ?